hydra-node/src/Hydra/Crypto.hs

```diff
@@ -121,24 +171,20 @@ instance FromJSON VerificationKey where
         (pure . HydraVerificationKey)
         . rawDeserialiseVerKeyDSIGN
 
-instance Arbitrary VerificationKey where
-  arbitrary = deriveVerificationKey . generateSigningKey <$> arbitrary
+instance HasTextEnvelope (VerificationKey HydraKey) where
+  textEnvelopeType _ =
+    "HydraVerificationKey_"
+      <> fromString (algorithmNameDSIGN (Proxy :: Proxy SignAlg))
 
--- | Serialise the verification key material as raw bytes.
-serialiseVerificationKeyToRawBytes :: VerificationKey -> ByteString
-serialiseVerificationKeyToRawBytes (HydraVerificationKey vk) = rawSerialiseVerKeyDSIGN vk
-
--- | Deserialise a verirfication key from raw bytes.
-deserialiseVerificationKeyFromRawBytes :: MonadFail m => ByteString -> m VerificationKey
-deserialiseVerificationKeyFromRawBytes bytes =
-  case rawDeserialiseVerKeyDSIGN bytes of
-    Nothing -> fail "failed to deserialise verification key"
-    Just key -> pure $ HydraVerificationKey key
-
--- | Get the Blake2b hash of a 'VerificationKey'.
-hashVerificationKey :: VerificationKey -> Hash Blake2b_256 VerificationKey
-hashVerificationKey (HydraVerificationKey vk) =
-  castHash $ hashVerKeyDSIGN vk
+-- | Create a new 'SigningKey' from a 'ByteString' seed. The created keys are
+-- not random and insecure, so don't use this in production code!
+generateSigningKey :: ByteString -> SigningKey HydraKey
+generateSigningKey seed =
+  HydraSigningKey . genKeyDSIGN $ mkSeedFromBytes padded
+ where
+  needed = fromIntegral $ seedSizeDSIGN (Proxy :: Proxy SignAlg)
+  provided = BS.length seed
+  padded = seed <> BS.pack (replicate (needed - provided) 0)
```

Comment:
That is slightly odd :thinking: ...

Reply:
This is already on `master` :P But it's currently getting fixed by @ffakenz to not pad, but hash the seed for a consistent key derivation.